sdk/python/kfp/dsl/_container_op.py

```diff
@@ -249,6 +170,132 @@ def add_env_variable(self, env_variable):
     self.env_variables.append(env_variable)
     return self
 
+
+class SideCar(ContainerBase):
+  """Represents a sidecar container to be provisioned together with a docker container image for ContainerOp."""
+
+  def __init__(self, name: str, image: str, command: str=None, arguments: str=None):
+    """
+    Args:
+      name: the name of the sidecar.
+      image: the sidecar container image name, such as 'rabbitmq:latest'
+      command: the command to run in the container.
+          If None, uses default CMD in defined in container.
+      arguments: the arguments of the command. The command can include "%s" and supply
+          a PipelineParam as the string replacement. For example, ('echo %s' % input_param).
+          At container run time the argument will be 'echo param_value'.
+    """
+    # name will be generated when attached to ContainerOps
+    super().__init__(name, image)
+    self.command = command
+    self.arguments = arguments
+
+    matches = []
+    for arg in (command or []) + (arguments or []):
+      match = re.findall(r'{{pipelineparam:op=([\w\s_-]*);name=([\w\s_-]+);value=(.*?)}}', str(arg))
+      matches += match
+
+    self.argument_inputs = [_pipeline_param.PipelineParam(x[1], x[0], x[2])
+                            for x in list(set(matches))]
+
+
+class ContainerOp(ContainerBase):
+  """Represents an op implemented by a docker container image."""
```

Comment:
Given docker is being phased out, may make sense to just call these 'container image'

Reply:
> Given docker is being phased out, may make sense to just call these 'container image'

Can you please provide some link documenting that?

Reply:
well @Ark-kun - its phased out from IBM`s official Kubernetes service 
https://www.ibm.com/blogs/bluemix/2019/01/ibm-cloud-kubernetes-service-supports-containerd/

Other providers are also discussing moving away - btw GCP lauched this in beta in Nov as well
https://cloud.google.com/blog/products/containers-kubernetes/containerd-available-for-beta-testing-in-google-kubernetes-engine